src/app/modules/onboarding/view.nim

```diff
@@ -13,3 +17,72 @@ QtObject:
     new(result, delete)
     result.QObject.setup
     result.delegate = delegate
+
+  proc syncStateChanged*(self: View) {.signal.}
+  proc getSyncState(self: View): int {.slot.} =
+    return self.syncState
+  QtProperty[int] syncState:
+    read = getSyncState
+    notify = syncStateChanged
+  proc setSyncState(self: View, syncState: int) =
+    self.syncState = syncState
+    self.syncStateChanged()
+
+  proc keycardStateChanged*(self: View) {.signal.}
+  proc getKeycardState(self: View): int {.slot.} =
+    return self.keycardState
+  QtProperty[int] keycardState:
+    read = getKeycardState
+    notify = keycardStateChanged
+  proc setKeycardState(self: View, keycardState: int) =
+    self.keycardState = keycardState
+    self.keycardStateChanged()
+
+  proc keycardRemainingPinAttemptsChanged*(self: View) {.signal.}
+  proc getKeycardRemainingPinAttempts(self: View): int {.slot.} =
+    return self.keycardRemainingPinAttempts
+  QtProperty[int] keycardRemainingPinAttempts:
+    read = getKeycardRemainingPinAttempts
+    notify = keycardRemainingPinAttemptsChanged
+  proc setKeycardRemainingPinAttempts(self: View, keycardRemainingPinAttempts: int) =
```

Comment:
This should probably not have a setter?

Reply:
(same for some of the other props in this file too)

Reply:
Those are not `slot`s , so they are for the module to update the properties if needed. I now realize I didn't make them public so it doesn't make sense as is.

It's possible that they are not needed though, I have yet to get the updated keycard code, but I think we'll need it for when an attempt is made, since we'll get a signal or return from the service that needs to update it.

I'll fix at least to make it public so the module can have access.